lib/std/jsfetch.nim

```diff
@@ -0,0 +1,203 @@
+        cache = "no-cache".cstring,
+        referrerPolicy = "no-referrer".cstring,
+        keepalive = false,
+        redirect = "follow".cstring,
+        referrer = "client".cstring,
+        integrity = "".cstring
+      )
+      doAssert options0.keepalive == false
+      doAssert options0.metod == "POST".cstring
+      doAssert options0.body == """{"key": "value"}""".cstring
+      doAssert options0.mode == "no-cors".cstring
+      doAssert options0.credentials == "omit".cstring
+      doAssert options0.cache == "no-cache".cstring
+      doAssert options0.referrerPolicy == "no-referrer".cstring
+      doAssert options0.redirect == "follow".cstring
+      doAssert options0.referrer == "client".cstring
+      doAssert options0.integrity == "".cstring
+      doAssert options0.toCstring is cstring
+
+    block:
+      let options1: FetchOptions = newFetchOptions(
+        metod =  HttpPost,
+        body = """{"key": "value"}""".cstring,
+        mode = fmNoCors,
+        credentials = fcOmit,
+        cache = fchNoCache,
+        referrerPolicy = frpNoReferrer,
+        keepalive = false,
+        redirect = frFollow,
+        referrer = "client".cstring,
+        integrity = "".cstring
+      )
+      doAssert options1.keepalive == false
+      doAssert options1.metod == $HttpPost
+      doAssert options1.body == """{"key": "value"}""".cstring
+      doAssert options1.mode == $fmNoCors
+      doAssert options1.credentials == $fcOmit
+      doAssert options1.cache == $fchNoCache
+      doAssert options1.referrerPolicy == $frpNoReferrer
+      doAssert options1.redirect == $frFollow
+      doAssert options1.referrer == "client".cstring
+      doAssert options1.integrity == "".cstring
+      doAssert options1.toCstring is cstring
+
+    block:
+      let response: Response = newResponse(body = "-. .. --".cstring)
+      doAssert response.clone() is Response ## Cloned.
+      doAssert response.toCstring is cstring
+      let request: Request = newRequest(url = "http://nim-lang.org".cstring)
+      doAssert request.clone() is Request   ## Cloned.
+      doAssert request.toCstring is cstring
+
+    if not defined(nodejs):
+      block:
+        proc doFetch(): Future[Response] {.async.} =
+          fetch "https://httpbin.org/get".cstring
+
+        proc example() {.async.} =
+          let response: Response = await doFetch()
+          doAssert response.ok
+          doAssert response.status == 200.cint
+          doAssert response.headers is Headers
+          doAssert response.body is Body
+          doAssert toCstring(response.body) is cstring
+
+        discard example()
+
+      when defined(nimExperimentalAsyncjsThen):
+        block:
+          proc example2 {.async.} =
+            await fetch("https://api.github.com/users/torvalds".cstring)
+              .then((response: Response) => response.json())
+              .then((json: JsObject) => console.log(json))
+              .catch((err: Error) => console.log("Request Failed", err))
```

Comment:
ditto: to strenghten the test, we need an assert on the content of json (eg: `assert json["login"] == torvalds` or similar